src/raftstore/store/peer_storage.rs

```diff
@@ -949,11 +1003,18 @@ impl PeerStorage {
         }
 
         if ctx.raft_state != self.raft_state {
-            try!(ctx.save_raft_to(&self.engine, &mut ready_ctx.wb));
+            try!(ctx.save_raft_state_to(&mut ready_ctx.raft_wb));
+            if snapshot_index > 0 {
+                // in case of restart happen when we just write region state to Applying,
```

Comment:
??? Please explain the exactly time serious when the `snapshot_index` is needed.